appservice/src/SiteClient.ts

```diff
@@ -456,13 +466,13 @@ export class SiteClient implements IAppSettingsClient {
         }));
     }
 
-    public async vfsPutItem(context: IActionContext, data: string | ArrayBuffer, url: string, rawHeaders?: {}): Promise<AzExtPipelineResponse> {
+    public async vfsPutItem(context: IActionContext, data: string | ArrayBuffer | Uint8Array, url: string, rawHeaders?: Record<string, string>): Promise<AzExtPipelineResponse> {
         const client: ServiceClient = await createGenericClient(context, this._site.subscription);
         const headers = createHttpHeaders(rawHeaders);
         return await client.sendRequest(createPipelineRequest({
             method: 'PUT',
             url,
-            body: typeof data === 'string' ? data : data.toString(),
+            body: typeof data === 'string' ? data : new TextDecoder('utf-8').decode(data),
```

Comment:
The old code worked, but somewhat by luck. VSCode's FS interfaces pass around `Uint8Array`, which is what we are passing into this method. At runtime, in the desktop VSCode, it's actually a Node `Buffer`, which implements `.toString(encoding?: ...)`, with UTF-8 being the default encoding. So, in the past we were taking the byte array and encoding it to UTF-8 before upload.

Thing is, `Uint8Array` (the base type also shared by web) does _not_ do that; it has just `.toString()` which writes unencoded, comma-separated `Uint8`s.

The better solution is to use `TextDecoder` to be explicit about what we're trying to do. It supports `ArrayBuffer`, `Uint8Array`, and `Buffer` all the same. `TextDecoder` is more technically correct, the best kind of correct.